OpenBSD: Use $MAKE in tests.

mjkillough opened the pull request:
These are two small changes to allow building on OpenBSD:

 - `#include <sys/wait.h>` in order to use `WEXITSTATUS`. This include is necessary on OpenBSD and is unlikely to hurt on other platforms, although I have not tested.
 - Use `$MAKE` in the regression tests driver rather than hard-coding to `make`. This allows it to be overridden with `MAKE=gmake` on BSD. I wasn't sure whether this was correct, or whether I should be using `$VERILATOR_MAKE`, which is also referenenced in `driver.pl`. As far as I could tell, `$VERILATOR_MAKE` serves a different purpose.

These are in separate commits, in case you'd like them to be separated out into separate PRs. Please let me know.

I've added myself to `CONTRIBUTORS` as specified in the contribution guide, to certify my contribution is my own.

gezalore commented on src/V3Os.cpp:
```diff
@@ -43,6 +43,7 @@
 #include <memory>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/wait.h>
```
Thanks for the PR, this actually is now included 12 lines below. I wonder why that did not get picked up (see #2353)

mjkillough replied:
Oh! That's my mistake for not checking master. Apologies - I was working off the `v4.034` tag, as I was trying to update the OpenBSD package to use that version. I can remove this commit. :-)

wsnyder commented:
Thanks for your first commit, this makes sense.  I have no objection to merging once the tests pass (note codacy seems broken this weekend so ignore that).

But, I see a lot of other tests and places where make is hardcoded in the test.  Perhaps these also need to be cleaned up and if so you can add them to your pull?  One method would be near the top of driver.pl in the BEGIN that sets VERILATOR_ROOT to set "$ENV{MAKE} ||= 'make'" then replace all the usages with $ENV{MAKE}?

```
test_regress/driver.pl:376:                 my $makecmd = $ENV{VERILATOR_MAKE} || "make$j &&";
test_regress/driver.pl:1130:                        cmd => ["make",
test_regress/driver.pl:1428:    my $ver = `make --version`;
test_regress/driver.pl:2234:    $_Cxx_Version ||= `make -C $ENV{VERILATOR_ROOT}/test_regress -f Makefile print-cxx-version`;
test_regress/driver.pl:2240:    $_Cfg_With_Threaded ||= `make -C $ENV{VERILATOR_ROOT} -f Makefile print-cfg-with-threaded`;
test_regress/t/t_dist_install.pl:28:    run(cmd => ["cd $root && make DESTDIR=$destdir install-all"],
test_regress/t/t_dist_install.pl:38:    run(cmd => ["cd $root && make DESTDIR=$destdir uninstall"],
test_regress/t/t_flag_csplit_off.pl:33:        cmd=>["make",
test_regress/t/t_embed1.pl:32:                "make", "-f".getcwd()."/Makefile_obj",
test_regress/t/t_prot_lib.pl:44:        cmd=>["make",
test_regress/t/t_dist_manifest.pl:86:    `cd $root && make dist-file-list`;
test_regress/t/t_dist_manifest.pl:87:    my $manifest_files = `cd $root && make dist-file-list`;
test_regress/t/t_a4_examples.pl:17:    run(cmd=>["make -C $example"]);
test_regress/t/t_dist_whitespace.pl:83:    `cd $root && make dist-file-list`;
test_regress/t/t_dist_whitespace.pl:84:    my $manifest_files = `cd $root && make dist-file-list`;
test_regress/t/t_prot_lib_secret.pl:24:run(cmd=>["make",
test_regress/t/t_var_in_assign_bad.pl:19:(!-d "obj_lint_only") or error("%Error: lint-only shouldn't make output directory");
test_regress/t/t_prot_lib_clk_gated.pl:46:        cmd=>["make",
test_regress/t/t_flag_csplit.pl:31:        cmd=>["make",
```

wsnyder commented:
P.S. Yes, VERILATOR_MAKE is a different purpose.

mjkillough commented:
Thanks for the quick response!

As mentioned in my comment above, I can probably remove my first commit for `<sys/wait.h>`. I'd missed that it'd been fixed already when applying my patch onto master.

>  Perhaps these also need to be cleaned up and if so you can add them to your pull?

Sure! I can have a go at this tomorrow. It is interesting that `make test` worked for me without those changes... I'll investigate that too.

wsnyder commented:
Ok, thanks. It might be that a 'make' under a submake works ok -- but if that was the case not sure why driver.pl wouldn't also work.

mjkillough commented:
I've updated this and it's passing in CI now. I've removed the commit to include `<sys/wait.h>` as it's no longer necessary.

A lot of the tests were skipped when I was running them locally yesterday, which is why I hadn't spotted the other uses of `make`.

wsnyder commented:
Great, thanks for updating.  BTW if you want to help keep this building maybe you'd like to add FreeBSD to our cron regressions?  #2390.
